tasks/emoji-unicode-version-map/index.js

```diff
@@ -37,83 +10,33 @@ const aliasMap = {
 
 
 function generateEmojiUnicodeVersionMap() {
-	const addingEmojipediaUnicodeVersionPromises = Object.keys(emojipediaUnicodeVersionUrlMap).map((unicodeVersionKey) => {
-		console.log(`Working on ${unicodeVersionKey}...`);
-		const emojipediaUrl = emojipediaUnicodeVersionUrlMap[unicodeVersionKey];
-		const cacheFileName = sanitizeForFileName(emojipediaUrl);
-		const cacheFilePath = `./cached-pages/${cacheFileName}.html`;
-		// Try to read from the cache
-		return readFile(cacheFilePath, 'utf8')
-			// Fallback to fetching the page
-			.catch(() => {
-				return fetch(emojipediaUrl)
-					.then((res) => res.text())
-					.then((html) => {
-						// Save the file as a cache
-						outputFile(cacheFilePath, html);
-						return html;
-					});
-			})
-			.then((html) => {
-				scrapeEmojipediaHtml(emojiMap, html, unicodeVersionKey);
-			});
-	});
+	const emojiToUnicodeVersionMap = {};
 
-	// Save the transformed emoji map
-	const fillingZwjSequencesPromise = Promise.all(addingEmojipediaUnicodeVersionPromises)
-		.then(() => {
-			console.log('Filling out ZWJ sequences...');
-			fillZwjSequenceUnicodeVersions(emojiMap);
-		});
+	Object.keys(emojioneMap).forEach((emojioneKey) => {
+		const emoji = emojioneMap[emojioneKey];
 
-	// Setup the legacy aliases
-	const addingAliasesPromise = fillingZwjSequencesPromise
-		.then(() => {
-			Object.keys(aliasMap).forEach((aliasName) => {
-				const aliasReferenceName = aliasMap[aliasName];
-				emojiMap[aliasName] = Object.assign({}, emojiMap[aliasReferenceName]);
-			});
-		});
+		const emojiShortName = emoji.shortname;
+		const emojiNameKey = emojiShortName.slice(1, emojiShortName.length - 1);
 
-	const ensureAllPopulatedPromise = addingAliasesPromise
-		.then(() => {
-			// Verify each emoji has a `unicodeVersion`
-			let missingVersionCount = 0;
-			let notStringVersionCount = 0;
-			Object.keys(emojiMap).forEach((emojiNameKey) => {
-				const emoji = emojiMap[emojiNameKey];
-				if(emoji.unicodeVersion === undefined) {
-					console.log(`Missing unicodeVersion property on \`${emojiNameKey}\``);
-					missingVersionCount += 1;
-				}
-				if(typeof emoji.unicodeVersion !== 'string') {
-					console.log(`Invalid unicodeVersion property, must be string on \`${emojiNameKey}\``);
-					notStringVersionCount += 1;
-				}
-			});
-			if(missingVersionCount > 0) {
-				throw new Error(`Missing unicodeVersion property on ${missingVersionCount} emojis`);
-			}
-			if(notStringVersionCount > 0) {
-				throw new Error(`unicodeVersion property not a string for ${notStringVersionCount} emojis`);
-			}
-		});
```

Comment:
I think we should keep this verification step

Reply:
the way the code is working now does not ever allow having values without a version, as we don't combine multiple sources each providing some versions, but we directly get it from emojione. Do you want to account for cases where the version is missing in emojione itself ?

Reply:
OK, re-adding the verification step, as there is actually a way to produce a violation: having an alias pointing to an non-existent name.